src/scripts/Sitemap.js

```diff
@@ -155,10 +155,20 @@ export default class Sitemap {
 	updateSelector(selector, selectorData) {
 		// selector is undefined when creating a new one and delete old one, if it exist
 		if (selector === undefined || selector.type !== selectorData.type) {
+			var CanHaveChilds = selector.canHaveChildSelectors()	
 			if (selector) {
```

Comment:
I would suggest change this block of code  (to the line 172)  and move some of its parts into a separate method that will receive the ID of the old and new selectors as input, and then replace the ID of the old selector with the ID of the new selector in all direct child selectors